### PR TITLE
fix vm blockchain.delBlock parameter

### PR DESCRIPTION
**Pull request by kumavis**

### CODE REVIEW

File: lib/runBlockchain.js [hunk 1]
```diff
@@ -52,7 +52,8 @@ module.exports = function (blockchain, cb) {
       }, function (err, results) {
         if (err) {
           // remove invalid block
-          self.stateManager.blockchain.delBlock(block, cb)
+          console.error(err)
```
Comment (axic) [inline lib/runBlockchain.js]: Please remove the log

**Comment by kumavis** [comment]
We need an optional logging mechanism -- the error here could be anything but we discard it and just attempt to delete the block. In my case the error was an important piece of evidence in tracking down a bug, but would have lost it..

any idea on optional logging mechanisms? just check `self.debug` ?

**Comment by axic** [comment]
Having a logging interface would be nice, but `console.log` probably isn't the best way to go in a library.

For starters it could be just an optional `log` in `self`, which is ignored by default, unless the user sets it to a method?

**Comment by wanderer** [comment]
recap on a conversation with @kumavis ; I think he can get all the information need from vm.on('block', cb) instead of built in loggin.

**Comment by wanderer** [comment]
@kumavis will merge! just take out that console.log :)

**Comment by kumavis** [comment]
@wanderer removed

**Pull request merged**